File: binary_search/bin_search.cpp
#include <bits/stdc++.h>

using std::cin;
using std::cout;
using std::endl;

int binSearch(int sortedArr[], int low, int high, int term){

    int idx = 0, mid = 0;

    if (low <= high) {

        mid = low + (high - low)/2;
    
        if (term == sortedArr[mid]) {
            return mid;

        } else if (term > sortedArr[mid]) {
            return binSearch(sortedArr, mid+1, high, term);
        }

        return binSearch(sortedArr, low, mid-1, term);
    } else {
        return -1;
    }
}

int main() {
    int arrSize = 0, totalNumbers = 0, num = 0;

    cin >> arrSize;

    int sortedArr[arrSize];

    for (int i = 0; i < arrSize; i++) {
        cin >> sortedArr[i];
    }

    cin >> totalNumbers;

    for (int j = 0; j < totalNumbers; j++) {
        cin >> num;

        int result = binSearch(sortedArr, 0, arrSize - 1, num);

        result != -1 ? cout << result << endl : cout << 'x' << endl;
    }

    return 1;
}

File: algorithms/binary_search/bin_search.py
from typing import Union

def bin_search(sorted_list: list, term: Union[str, int], low: int, high: int, index_offset: int = 0) -> Union[tuple, int]:

    if low <= high:
        mid = (low + high) // 2
        
        if term == sorted_list[mid]:
            return mid+index_offset, sorted_list[mid]
        
        elif term > sorted_list[mid]:
            return bin_search(sorted_list, term, mid+1, high, index_offset)
        
            
        return bin_search(sorted_list, term, low, mid-1, index_offset)
        
    else:
        return -1

